txp/Pos/Txp/Settings/Global.hs

```diff
@@ -45,8 +47,22 @@ type TxpGlobalApplyMode ctx m =
 type TxpGlobalRollbackMode m = TxpCommonMode m
 
 -- [CSL-1156] Maybe find better approach (at least wrap into normal types).
-type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
-type TxpBlund = (TxpBlock, TxpUndo)
+-- type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
+-- type TxpBlund = (TxpBlock, TxpUndo)
+
+-- | Representation of 'Block' passed to a component.
+data ComponentBlock payload =
```

Comment:
This type should be in `cardano-sl-core`, because it's needed for `cardano-sl-update`, `cardano-sl-txp`, etc.
`Pos.Core.Block.Union.Types` is likely the right place.